packages/sdk-client/src/utils/authorization.helper.ts

```diff
@@ -0,0 +1,151 @@
+};
+
+export const calculateSignature = (secret: string, stringToSign: string): string => {
+  // Signature = Base64 ( HMAC-SHA256 ( Base64-Decode ( ApplicationSecret ), UTF8 ( StringToSign ) ) );
+  return crypto.createHmac('sha256', Buffer.from(secret, 'base64'))
+    .update(Buffer.from(stringToSign, 'utf-8'))
+    .digest('base64');
+};
+
+const buildStringToSign = (
+  httpVerb: string,
+  contentMD5: string,
+  contentType: string,
+  canonicalizedHeaders: string,
+  canonicalizedResource: string,
+): string => {
+  return `${httpVerb}\n${contentMD5}\n${contentType}\n${canonicalizedHeaders}\n${canonicalizedResource}`;
+};
+
+export const generateAuthorizationHeader = (
+  httpVerb: string,
+  body: RequestBody | undefined,
+  contentType: string,
+  canonicalizedHeaders: string,
+  canonicalizedResource: string,
+  applicationKey: string,
+  applicationSecret: string,
+): string => {
+  const hasBody = !!body;
+  const stringToSign = buildStringToSign(
+    httpVerb,
+    hasBody ? calculateMD5(body as string) : '',
+    contentType,
+    canonicalizedHeaders,
+    canonicalizedResource,
+  );
+
+  const signature = calculateSignature(applicationSecret, stringToSign);
+  return `Application ${applicationKey}:${signature}`;
+};
+
+export const validateAuthenticationHeader = (
+  applicationKey: string,
+  applicationSecret: string,
+  headers: IncomingHttpHeaders,
+  path: string,
+  body: any,
+  method: string,
+): boolean => {
+  const normalizedHeaders = Object.fromEntries(
+    Object.entries(headers)
+      .map(([key, value]) => [key.toLowerCase(), value])
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      .filter(([_, value]) => value !== undefined),
+  ) as { [p: string]: string | string[] };
+
+  const authorization = getHeader(normalizedHeaders.authorization);
+  const authParts = checkAuthorizationHeaderFormat(authorization);
+
+  const authorizationScheme = authParts[0].toLowerCase();
+  const authorizationValue = authParts[1];
+  if (authorizationScheme === 'basic') {
+    return validateBasicAuth(authorizationValue, applicationKey, applicationSecret);
+  }
+  if (authorizationScheme === 'application') {
+    return validateApplicationAuth(
+      authorizationValue,
+      normalizedHeaders,
+      applicationKey,
+      applicationSecret,
+      path,
+      body,
+      method,
+    );
+  }
+  // Other schemes than 'basic' or 'application' are not supported
+  throw new Error(`Scheme is not valid: ${authParts[0]}`);
+};
+
+const validateApplicationAuth = (
+  authorizationValue: string,
+  normalizedHeaders: {[p: string]: string | string[]},
+  applicationKey: string,
+  applicationSecret: string,
+  path: string,
+  body: any,
+  method: string,
+): boolean => {
+  const authKeyAndSecret = authorizationValue.split(':');
+  if(authKeyAndSecret.length !== 2) {
+    throw new Error('Invalid authorization value format provided');
```

Comment:
From a "consumer" point of view, is it better to
- return `false` from `validateAuthenticationHeader` call because of invalid authentication ?
- or throw an exception ?

Currently the exception is not dedicated to an auth issue.
So even if catched from application/consumer; it won't enable to identified an auth issue vs another one.

Without real useCase, my personal feeling is to let final application handle not-authentificated feature itself by:
- adding here a log about the error
- return false without exception... 

but a `documented` and `dedicated` Exception could do the job too...

Reply:
Fixed, the is now returning a boolean